hydroflow_plus/src/singleton.rs

```diff
@@ -134,24 +137,24 @@ impl<'a, T, N: Location<'a>> DeferTick for Singleton<T, Bounded, Tick<N>> {
 
 impl<'a, T, N: Location<'a>> CycleComplete<'a, TickCycle> for Singleton<T, Bounded, Tick<N>> {
     fn complete(self, ident: syn::Ident) {
-        self.flow_state.borrow_mut().leaves.as_mut().expect("Attempted to add a leaf to a flow that has already been finalized. No leaves can be added after the flow has been compiled.").push(HfPlusLeaf::CycleSink {
+        self.flow_state().clone().borrow_mut().leaves.as_mut().expect("Attempted to add a leaf to a flow that has already been finalized. No leaves can be added after the flow has been compiled.").push(HfPlusLeaf::CycleSink {
             ident,
-            location_kind: self.location_kind,
+            location_kind: self.location_kind(),
             input: Box::new(self.ir_node.into_inner()),
```

Comment:
Could avoid the clone via
```rust
input: Box::new(self.ir_node.replace(HfPlusNode::Placeholder)),
```
(or implement `Default` for `HfPlusNode` -> `Placeholder` and use `self.ir_node.take()`)

Doesn't really matter that much though